integrationTests/testNetwork.go

```diff
@@ -0,0 +1,107 @@
+package integrationTests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/stretchr/testify/require"
+)
+
+type TestNetwork struct {
+	NumShards         int
+	NodesPerShard     int
+	NodesInMetashard  int
+	Nodes             []*TestProcessorNode
+	Proposers         []int
+	Advertiser        p2p.Messenger
+	AdvertiserAddress string
+	Round             uint64
+	Nonce             uint64
+	T                 *testing.T
+}
+
+func NewTestNetwork(t *testing.T) *TestNetwork {
+	// TODO replace testing.T with testing.TB everywhere in the integrationTest utilities
+	return &TestNetwork{
+		T: t,
+	}
+}
+
+func (net *TestNetwork) Start() {
+	net.createAdvertiser()
+	net.createNodes()
+	net.indexProposers()
+	net.startNodes()
+
+	net.Round = 0
+	net.Nonce = 0
+}
+
+func (net *TestNetwork) Step() {
```

Comment:
add comment for exported public functions

Reply:
Right, missed that. :D

Reply:
Added.